Fixed grammar and sentence structure (#49)
eoxhub-workspaces/documentation: applications/argo/capability_integration.md

# Integration Requirements for Argo Workflows on EOxHub

This document is intended as support material describing how to provide a capability implementation and its description so that EOX can integrate it as a workflow within an EOxHub Workspace. This includes input definitions, expected Docker image standards, and output formats.

## Input Requirements

To ensure compatibility with various interfaces (e.g. OGC Process API) and allow on-demand execution of the capability, some input requirements need to be considered. Furthermore, additional information is needed regarding the following points:

- Description of expected input data - does the service expect data already present or data are downloaded as part of the dockerized algorithm?
- Description of expected arguments and parameters
- Description of expected output - see Output requirements section

### Arguments and parameters

The eodash processing widget supports multiple ways of passing input. 

> - Area/location — the process can take a drawn point or polygon as input from the eodash user interface. For this integration, the input field must accept either coordinates directly or GeoJSON as a string. File input is not accepted.

Reviewer: ```suggestion
- Area/location - the process can take a drawn point or polygon as input from the eodash user interface. For this integration, the input field must accept either coordinates directly or GeoJSON as a string. File input is not accepted.
```

    - Example GeoJSON Feature String '{"type":"Feature","geometry":{"type":"Polygon" "coordinates":[[[30,10],[40,40],[20,40],[10,20],[30,10]]]},"properties":{}}'

> - Date — standard HTML date formats are supported. eodash also supports start and end times to create a range. 

Reviewer: ```suggestion
- Date - standard HTML date formats are supported. eodash also supports start and end times to create a range. 
```

    - "YYYY-MM-DD" e.g. 2015-05-30 for date.
    - "YYYY-MM-DDThh:mm" for datetime 2025-07-02T06:33 
- Numeric fields - integer or float values
- Text fields
- Dropdowns with limited options
  

All fields can have a default value

## Docker Image Requirements

Your process must be encapsulated in a Docker image. There are no strict limitations of what can be done, but there are good practices that help integration:

> - The image is "slim" — only the required dependencies are installed.

Reviewer: ```suggestion
- The image is "slim" - only the required dependencies are installed.
```

- The image is tagged with a version based on [semantic versioning](https://semver.org/).
- The algorithm logs to standard output (stdout) helping to debug potential issues.

> - Avoid sidecars, Docker-in-Docker (DinD), and similar additional services where possible.

Reviewer: ```suggestion
- Avoid sideloading/sidecars (Docker in Docker) and similar additional services where possible to keep the structure simple
```
